homeassistant/components/neato/config_flow.py

```diff
@@ -1,112 +1,62 @@
+        return await self.async_step_user()
+
+    async def async_oauth_create_entry(self, data: dict) -> dict:
+        """Create an entry for the flow. Update an entry if one already exist."""
+        current_entries = self._async_current_entries()
+        if current_entries and current_entries[0].version == 1:
+            # Update version 1 to 2
+            current_entries[0].version = self.VERSION
+            self.hass.config_entries.async_update_entry(
+                current_entries[0], title=self.flow_impl.name, data=data
             )
-            if error:
-                errors["base"] = error
-            else:
-                return self.async_create_entry(
-                    title=user_input[CONF_USERNAME],
-                    data=user_input,
-                    description_placeholders={"docs_url": DOCS_URL},
-                )
-
-        return self.async_show_form(
-            step_id="user",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(CONF_USERNAME): str,
-                    vol.Required(CONF_PASSWORD): str,
-                    vol.Optional(CONF_VENDOR, default="neato"): vol.In(VALID_VENDORS),
-                }
-            ),
-            description_placeholders={"docs_url": DOCS_URL},
-            errors=errors,
-        )
-
-    async def async_step_import(self, user_input):
-        """Import a config flow from configuration."""
-
-        if self._async_current_entries():
-            return self.async_abort(reason="already_configured")
-
-        username = user_input[CONF_USERNAME]
-        password = user_input[CONF_PASSWORD]
-        vendor = user_input[CONF_VENDOR]
-
-        error = await self.hass.async_add_executor_job(
-            self.try_login, username, password, vendor
-        )
-        if error is not None:
-            _LOGGER.error(error)
-            return self.async_abort(reason=error)
-
-        return self.async_create_entry(
-            title=f"{username} (from configuration)",
-            data={
-                CONF_USERNAME: username,
-                CONF_PASSWORD: password,
-                CONF_VENDOR: vendor,
-            },
-        )
-
-    @staticmethod
-    def try_login(username, password, vendor):
-        """Try logging in to device and return any errors."""
-        this_vendor = None
-        if vendor == "vorwerk":
-            this_vendor = Vorwerk()
-        else:  # Neato
-            this_vendor = Neato()
-
-        try:
-            Account(username, password, this_vendor)
-        except NeatoLoginException:
-            return "invalid_auth"
-        except NeatoRobotException:
-            return "unknown"
-
-        return None
+            # TODO: This doesn't work because the entry state is still migration_error
+            # self.hass.async_create_task(
+            #    self.hass.config_entries.async_reload(current_entries[0].entry_id)
+            # )
```

Comment:
How can we reset the status of the entry?

Reply:
Maybe we shouldn't use the migration, since we can't migrate the config entry automatically?

Reply:
I decided to keep the `version` attribute but dropped the migraton api. What do you think?